data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/RecordDispatcherImpl.java

```diff
@@ -157,60 +180,82 @@ private void onRecordReceived(final KafkaConsumerRecord<Object, CloudEvent> reco
       }
     }
 
-    recordDispatcherListener.recordReceived(record);
+    recordDispatcherListener.recordReceived(recordContext.getRecord());
     // Execute filtering
-    if (filter.test(record.value())) {
-      onFilterMatching(record, finalProm);
+    final var pass = filter.test(recordContext.getRecord().value());
+
+    Metrics
+      .eventProcessingLatency(this.resourceContext.getTags())
+      .register(meterRegistry)
+      .record(recordContext.performLatency());
+
+    // reset timer to start calculating the dispatch latency.
+    recordContext.resetTimer();
+
+    if (pass) {
+      onFilterMatching(recordContext, finalProm);
     } else {
-      onFilterNotMatching(record, finalProm);
+      onFilterNotMatching(recordContext, finalProm);
     }
   }
 
-  private void onFilterMatching(final KafkaConsumerRecord<Object, CloudEvent> record, final Promise<Void> finalProm) {
-    logDebug("Record match filtering", record);
-    subscriberSender.apply(record)
-      .onSuccess(res -> onSubscriberSuccess(record, finalProm))
-      .onFailure(ex -> onSubscriberFailure(record, finalProm));
+  private void onFilterMatching(final ConsumerRecordContext recordContext, final Promise<Void> finalProm) {
+    logDebug("Record match filtering", recordContext.getRecord());
+    subscriberSender.apply(recordContext.getRecord())
+      .onSuccess(response -> onSubscriberSuccess(response, recordContext, finalProm))
+      .onFailure(ex -> onSubscriberFailure(ex, recordContext, finalProm));
   }
 
-  private void onFilterNotMatching(final KafkaConsumerRecord<Object, CloudEvent> record,
+  private void onFilterNotMatching(final ConsumerRecordContext recordContext,
                                    final Promise<Void> finalProm) {
-    logDebug("Record doesn't match filtering", record);
-    recordDispatcherListener.recordDiscarded(record);
+    logDebug("Record doesn't match filtering", recordContext.getRecord());
```

Comment:
```suggestion
    logDebug("Record did not match filtering", recordContext.getRecord());
```

nit: to show that this is after the fact